bases/ember.json

```diff
@@ -26,16 +27,13 @@
     "noUncheckedIndexedAccess": true,
 
     // Interop: these are viral and will require anyone downstream of your
-    // package to *also* set them to true. If you *must* enable them to consume
-    // an upstream package, you should document that for downstream consumers to
-    // be aware of.
-    //
-    // These *are* safe for apps to enable, since they do not *have* downstream
-    // consumers; but leaving them off is still preferred when possible, since
-    // it makes it easier to switch between apps and addons and have the same
-    // rules for what can be imported and how.
-    "allowSyntheticDefaultImports": false,
-    "esModuleInterop": false,
+    // package to *also* set them to true. However, this represents the way that
+    // bundlers actually work, and is future-compatible with the closest module
+    // modes: "nodenext" in TS 4.7+ and "mixed" in 5.0+ mode. Since Ember apps
+    // and addons never emit with `tsc`, this is safe: it makes type-checking do
```

Comment:
@chriskrycho To your point about never emitting with `tsc`, I wonder if we should just drop `esModuleInterop` here entirely, since that setting is for controlling emit?

Reply:
That's a good point. Then the compiler option will default to whatever is implied by the `module` config. That in turn prompts another thought: I wonder if we should also update to use `"module": "nodenext"` here until we require TS 5.0, to get the most compatibility with Node and bundler behavior for now? That would *imply* `esModuleInterop: true` but not in a way we care about. 🤔

Reply:
We chatted in Discord DMs a bit and concluded: we should keep `module` as is for now (and do a breaking release with *that* change) because changing it would opt people into having `exports` start working, which would change the semantics of modules they consume from their POV.